### PR TITLE
Simulator api update

### LINKED ISSUE #2: Set formalism on simulation init
Currently, every incoming message needs a `formalism` object so that it can be passed to the right plugin and the plugin can choose the right formalism.

It would be much more efficient to set the formalism on simulation init for each simulation session.

### LINKED ISSUE #3: Add simulation initialized callback
The gateway should inform the client when the plugin has initialized the simulation.
This way the client can avoid sending further requests that can not be processed immediately.

### LINKED ISSUE #4: Add abstract plugin
An abstract plugin class can serve as an implementation guideline and also abstract some functionality like registering ServerManager handlers.

### PR DESCRIPTION
**Pull request by 0x4d**
Closes #4 
Closes #3 
Closes #2

### CODE REVIEW

File: src/core/SimulationManager.js [hunk 1]
```diff
@@ -8,38 +8,39 @@ class SimulationManager {
...
-                    .step();
+            socket.on('simulation.step', () => {
+                activePlugin && activePlugin.step();
             });
 
-            socket.on('simulation.stop', (formalism) => {
-                this.getPlugin(formalism.plugin)
-                    .stop();
+            socket.on('simulation.stop', () => {
+                activePlugin && activePlugin.stop();
```
Comment (tim-kilian) [inline src/core/SimulationManager.js:35]: Ahh, way better imo.

File: src/plugin/AbstractPlugin.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
+/**
+ * @abstract
+ */
+class AbstractPlugin {
```
Comment (tim-kilian) [inline src/plugin/AbstractPlugin.js:4]: I am not sure if the AbstractPlugin should be in the gataway.

**Review by tim-kilian — APPROVED** [review]

File: src/plugin/AbstractPlugin.js (hunk 2 above)
Comment (0x4d) [reply src/plugin/AbstractPlugin.js:4]: Yes could be in it's own package like renew-formalism :thinking:

Comment (tim-kilian) [reply src/plugin/AbstractPlugin.js:4]: Thats what I mean.

**Pull request merged**